packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,394 @@
+    final List<dynamic> rootChildren = apkAnalysisJson['children'] as List<dynamic>;
+    apkAnalysisJson.remove('children');
+    
+    apkAnalysisJson['type'] = 'apk';
+    apkAnalysisJson['children'] = rootChildren;
+
+    return apkAnalysisJson;
+  }
+
+  // Parse the output of unzip -v which shows the zip's contents' compressed sizes.
+  // Example output of unzip -v:
+  //  Length   Method    Size  Cmpr    Date    Time   CRC-32   Name
+  // --------  ------  ------- ---- ---------- ----- --------  ----
+  //    11708  Defl:N     2592  78% 00-00-1980 00:00 07733eef  AndroidManifest.xml
+  //     1399  Defl:N     1092  22% 00-00-1980 00:00 f53d952a  META-INF/CERT.RSA
+  //    46298  Defl:N    14530  69% 00-00-1980 00:00 17df02b8  META-INF/CERT.SF
+
+  final RegExp _parseUnzipOutput = RegExp(r'^\s*\d+\s+[\w|:]+\s+(\d+)\s+.*  (.+)$');
+
+  SymbolNode _parseUnzipFile(String unzipOut) {
+    final Map<List<String>, int> pathsToSize = <List<String>, int>{};
+
+    // Parse each path into pathsToSize so that the key is a list of
+    // path parts and the value is the size.
+    // For example:
+    // 'path/to/file' where file = 1500 => pathsToSize[['path', 'to', 'file']] = 1500
+    for (final String line in const LineSplitter().convert(unzipOut)) {
+      final RegExpMatch match = _parseUnzipOutput.firstMatch(line);
+      if (match == null) {
+        continue;
+      }
+      pathsToSize[match.group(2).split('/')] = int.parse(match.group(1));
+    }
+
+    final SymbolNode rootNode = SymbolNode('Root');
+
+    SymbolNode currentNode = rootNode;
+    for (final List<String> paths in pathsToSize.keys) {
+      for (final String path in paths) {
+        SymbolNode childWithPathAsName = currentNode.childByName(path);
+
+        if (childWithPathAsName == null) {
+          childWithPathAsName = SymbolNode(path);
+
+          if (path.endsWith('libapp.so')) {
+            childWithPathAsName.name += ' (Dart AOT)';
+          } else if (path.endsWith('libflutter.so')) {
+            childWithPathAsName.name += ' (Flutter Engine)';
+          }
+          currentNode.addChild(childWithPathAsName);
+        }
+        childWithPathAsName.addValue(pathsToSize[paths]);
+        currentNode = childWithPathAsName;
+      }
+      currentNode = rootNode;
+    }
+
+    return rootNode;
+  }
+
+  /// Prints the paths from currentNode all leaf nodes.
+  void _printLibDetails(
+    SymbolNode currentNode,
+    String totalPath,
+    File aotSizeJson,
+  ) {
+    totalPath += currentNode.name;
+
+    if (currentNode.children.isNotEmpty && !currentNode.name.contains('libapp.so')) {
+      for (final SymbolNode child in currentNode.children) {
+        _printLibDetails(child, totalPath + '/', aotSizeJson);
+      }
+    } else {
+      // Print total path and size if currentPath does not have any chilren.
+      _printEntitySize(totalPath, currentNode.value, 2);
+
+      const String libappPath = 'lib/arm64-v8a/libapp.so';
```

Comment:
add code comment here. e.g. we picked this specific file to match and print the aot content because it's likely the most popular architecture and other architecture sizes should be similar so it's not worth repeating

Reply:
I think snapshots for all three platforms will be included in a follow-up CL. Should I still add the comment?